### PR TITLE
Add typecasting from `ActionController::Parameters` to `Hash`

### PR DESCRIPTION
**Pull request by ojab**
Maybe specs should be moved to `spec/lib/active_data_spec.rb`, please speak up.

### CODE REVIEW

File: lib/active_data.rb [hunk 1]
```diff
@@ -59,6 +59,8 @@ def self.config
     case value
     when ::Hash then
       value
+    when defined?(ActionController::Parameters) && ActionController::Parameters
```
Comment (ojab) [inline lib/active_data.rb:62]: I could define typecasters inside `if defined?(ActionController::Parameters)` to save `defined?` call on each typecast, but I'm not sure that we really care about it here.

Comment (ojab) [reply lib/active_data.rb:60]: It's located before `::Hash` check because in rails-4.2 `ActionController::Parameters` inherits from `::Hash`.

Comment (pyromaniac) [reply lib/active_data.rb:60]: We probably do care about it, the less checks we have - the faster it is going to be, right?

File: .rubocop.yml [hunk 2]
```diff
@@ -49,6 +49,7 @@ Metrics/BlockLength:
 Metrics/ModuleLength:
   Exclude:
     - '**/*_spec.rb'
+    - 'lib/active_data.rb'
```
Comment (ojab) [inline .rubocop.yml:52]: Not sure how to handle it, all typecaster are already longer than threshold & moving them to the separate files [per class] is an overkill.

**Pull request merged**